### PR TITLE
1829 Implementation South England

### PR DESCRIPTION
**Pull request by volker18xxdev**
I've done the first steps for implement 1829 South England from Francis.
Map, Trains, Phases, Private companies, Tile-Definition (Some Definitions were wrong. When in doubt of use in other games, i duplicate the tile and add an 's' to the tile name.)

### CODE REVIEW

File: lib/engine/config/tile.rb [hunk 1]
```diff
@@ -30,6 +30,7 @@ module Tile
         '56' => 'town=revenue:10;town=revenue:10;path=a:0,b:_0;path=a:_0,b:2;path=a:1,b:_1;path=a:_1,b:3',
         '57' => 'city=revenue:20;path=a:0,b:_0;path=a:_0,b:3',
         '58' => 'town=revenue:10;path=a:0,b:_0;path=a:_0,b:2',
+        '59s' => 'city=revenue:20;path=a:0,b:_0',
```
Comment (roseundy) [inline lib/engine/config/tile.rb:33]: Game-specific tiles should not be put into this file. They should go into game/g_1829/map.rb

File: lib/engine/game/g_1829.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+# frozen_string_literal: true
+
+module Engine
+  module Game
+    module G1830
```
Comment (roseundy) [inline lib/engine/game/g_1829.rb:5]: G1829

File: lib/engine/game/g_1829/game.rb [hunk 3]
```diff
@@ -0,0 +1,496 @@
+# frozen_string_literal: true
+
+# todos
+# upgrade tile 3,4,1,2,55 to 12,13,14,15
+# graybrown tile
+# foreign tokened cities do not count for dividends
+# survey party implentation
+# steamship
+# close privates voluntary
+# fixed par price and fixed order in OR
```
Comment (roseundy) [inline lib/engine/game/g_1829/game.rb:10]: We usually create a tracking issue for the "to do" list.

File: lib/engine/game/g_1829/game.rb [hunk 4]
```diff
@@ -0,0 +1,496 @@
...
+        CURRENCY_FORMAT_STR = '$%dP'
+        GAME_END_CHECK = { bank: :immediate }.freeze
+
+        BANK_CASH = 20_000
+
+        CERT_LIMIT = { 3 => 18, 4 => 18, 5 => 17, 6 => 14, 7 => 12, 8 => 10, 9 => 9 }.freeze
+
+        STARTING_CASH = { 3 => 840, 4 => 630, 5 => 1504, 6 => 420, 7 => 360, 8 => 315, 9 => 280 }.freeze
+
+        TILES = {
```
Comment (roseundy) [inline lib/engine/game/g_1829/game.rb:40]: Put this into game/g_1829/map.rb

File: lib/engine/game/g_1829/game.rb [hunk 5]
```diff
@@ -0,0 +1,496 @@
...
+                  },
+                  {
+                    name: '7',
+                    distance: 7,
+                    price: 720,
+                    num: 4,
+                    events: [{ 'type' => 'close_companies' }, { 'type' => 'private' }],
+                  }].freeze
+
+        COMPANIES = [
```
Comment (roseundy) [inline lib/engine/game/g_1829/game.rb:212]: Put this into game/g_1829/entities.rb

File: lib/engine/game/g_1829/game.rb [hunk 6]
```diff
@@ -0,0 +1,496 @@
...
+            tokens: [0, 40, 100],
+            coordinates: 'L21',
+            type: 'init10',
+            color: :"#ADD8E6",
+            text_color: 'yellow',
+            reservation_color: nil,
+          },
+        ].freeze
+
+        HEXES = {
```
Comment (roseundy) [inline lib/engine/game/g_1829/game.rb:381]: Put this into game/g_1829/map.rb

File: g_1829.rb [hunk 7]
```diff
@@ -0,0 +1,8 @@
+# frozen_string_literal: true
+
+module Engine
+  module Game
+    module G1830
```
Comment (roseundy) [inline g_1829.rb:5]: G1829

Comment (volker18xxdev) [reply g_1829.rb:5]: done

File: lib/engine/config/tile.rb (hunk 1 above)
Comment (volker18xxdev) [reply lib/engine/config/tile.rb:33]: moved to g_1829/map.rb

File: g_1829.rb (hunk 7 above)
Comment (tobymao) [reply g_1829.rb:5]: i don't see any changes?

Comment (tobymao) [reply g_1829.rb:5]: still don't see any changes...

Comment (tobymao) [reply g_1829.rb:5]: this says 1830

**Comment by volker18xxdev** [comment]
I'll close this PR and create a new one. I can't find the broken file.

**Comment by volker18xxdev** [comment]
i find the file and deleted it. i'll reopen then PR

**Comment by volker18xxdev** [comment]
sorry, i found the mentioned file and deleted it. i'll update the other files.

**Review by tobymao — APPROVED** [review]

**Pull request merged**